sycl/source/detail/jit_compiler.cpp

```diff
@@ -316,7 +325,8 @@ static void resolveInternalization(ArgDesc &Arg, unsigned KernelIndex,
 
       if (PreviousDefinition.PromotionTarget == Promotion::Local) {
         // Recompute the local size with adapted promotion target.
-        auto ThisLocalSize = getLocalSize(NDRange, Req, Promotion::Local);
+        auto ThisLocalSize =
+            getLocalSize(NDRange, std::nullopt, Req, Promotion::Local);
```

Comment:
Same:

Would UserGlobalSize ever be non-std::nullopt here?

Reply:
Same rationale: If the current definition is in a RR kernel, then private i13n can't be upgrade to local.